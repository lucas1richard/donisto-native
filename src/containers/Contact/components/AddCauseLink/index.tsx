import * as React from 'react';
import NavigationService from 'containers/Navigation/NavigationService';
import routeNames from 'containers/Navigation/routeNames';
import Touchable from 'components/Touchable';
import { Entypo } from '@expo/vector-icons';
import { View } from 'react-native';
import causeStyle from '../Cause/styles';
import style from './styles';

function AddCauseLink() {
  return (
    <Touchable onPress={() => NavigationService.navigate(routeNames.CAUSE)}>
      <View style={[causeStyle.wrapper, style.wrapper]}>
        <View style={causeStyle.textWrapper}>
          <Entypo
            name="plus"
            size={50}
            color="#fff"
            style={{ marginRight: 5 }}
          />
        </View>
      </View>
    </Touchable>
  );
}

export default AddCauseLink;