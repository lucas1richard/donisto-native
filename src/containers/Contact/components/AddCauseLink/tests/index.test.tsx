import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddCauseLink from '..';

describe('<AddCauseLink />', () => {
  const renderComponent = () => toJson(shallow(<AddCauseLink />));
  it('renders with just text', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});