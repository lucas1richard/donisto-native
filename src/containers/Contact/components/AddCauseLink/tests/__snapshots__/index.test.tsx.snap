// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCauseLink /> renders with just text 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "height": 100,
          "marginRight": 12,
          "width": 100,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "justifyContent": "center",
          "marginRight": 12,
          "overflow": "hidden",
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.6)",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="plus"
        size={50}
        style={
          Object {
            "marginRight": 5,
          }
        }
      />
    </Component>
  </Component>
</TouchableOpacity>
`;
